packages/atomic-angular/projects/atomic-angular/src/lib/atomic-angular.module.ts

```diff
@@ -0,0 +1,34 @@
+import {CommonModule} from '@angular/common';
+import {ModuleWithProviders, NgModule} from '@angular/core';
+
+import {
+  AtomicSearchInterface,
+  AtomicSearchBox,
+  AtomicResultList,
+  AtomicLoadMoreResults,
+} from './stencil-generated/components';
+
+import {defineCustomElements} from '@coveo/atomic/loader';
+defineCustomElements(window);
+
+const DECLARATIONS = [
+  AtomicSearchInterface,
+  AtomicSearchBox,
+  AtomicResultList,
+  AtomicLoadMoreResults,
+];
+
+@NgModule({
```

Comment:
So that's the thing that can workaround the problem with angular ivy:.

For example doing this in the consuming app: 
```
import {AtomicSearchInterface, AtomicSearchBox} from '@coveo/atomic-angular';

@NgModule({
  declarations: [AppComponent],
  imports: [BrowserModule, AppRoutingModule, AtomicSearchInterface, AtomicSearchBox],
  providers: [],
  bootstrap: [AppComponent],
})
export class AppModule {}
```
Then we get errors like this at runtime: 

```
Error: ../ui-kit/packages/atomic-angular/dist/atomic-angular/lib/stencil-generated/components.d.ts:478:22 - error NG6002: Appears in the NgModule.imports of AppModule, but could not be resolved to an NgModule class.

This likely means that the dependency (@coveo/atomic-angular) which declares AtomicSearchInterface has not been processed correctly by ngcc.

478 export declare class AtomicSearchInterface {
```

However, in the consuming app if we do this: 
```
import { AtomicAngularModule } from '@coveo/atomic-angular';
@NgModule({
  declarations: [AppComponent],
  imports: [BrowserModule, AppRoutingModule, AtomicAngularModule],
  providers: [],
  bootstrap: [AppComponent],
})
export class AppModule {}
```

Then it works correctly. The catch is that we need to have a static declaration in the `AtomicAngularModule`. 

I think we can workaround this by changing the angular output plugin to generate that array at compile time, so that we don't need to manually update it.


And then, apparently the best practice in angular world is a Single Component Angular Module (SCAM), which means the best would be for us to have one module for each component, instead of one module that contains every component : https://angular-training-guide.rangle.io/modules/module-scam-pattern#what-is-the-scam-single-component-angular-module-pattern-in-angular


With that optic, I'm thinking of submitting a PR to the stencil team to add the boiler plate needed to the generated files.

Reply:
I'm also suspecting a weird symlink issue with Angular, which might not require us to go a too complicated approach 🤞